gtars-wasm/src/regionset.rs

```diff
@@ -0,0 +1,159 @@
+use gtars_genomicdist::statistics::GenomicIntervalSetStatistics;
+use wasm_bindgen::prelude::*;
+
+use crate::models::BedEntries;
+
+#[wasm_bindgen(js_name = "ChromosomeStatistics")]
+#[derive(serde::Serialize)]
+pub struct JsChromosomeStatistics {
+    chromosome: String,
+    number_of_regions: u32,
+    minimum_region_length: u32,
+    maximum_region_length: u32,
+    mean_region_length: f64,
+    median_region_length: f64,
+    start_nucleotide_position: u32,
+    end_nucleotide_position: u32,
+}
+
+#[wasm_bindgen(js_class = "ChromosomeStatistics")]
+impl JsChromosomeStatistics {
+    #[wasm_bindgen(getter)]
+    pub fn chromosome(&self) -> String {
+        self.chromosome.clone()
+    }
+    #[wasm_bindgen(getter)]
+    pub fn start_nucleotide_position(&self) -> u32 {
+        self.start_nucleotide_position
+    }
+    #[wasm_bindgen(getter)]
+    pub fn end_nucleotide_position(&self) -> u32 {
+        self.end_nucleotide_position
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn number_of_regions(&self) -> u32 {
+        self.number_of_regions
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn minimum_region_length(&self) -> u32 {
+        self.minimum_region_length
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn maximum_region_length(&self) -> u32 {
+        self.maximum_region_length
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn mean_region_length(&self) -> f64 {
+        self.mean_region_length
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn median_region_length(&self) -> f64 {
+        self.median_region_length
+    }
+}
+
+#[wasm_bindgen(js_name = "RegionDistribution")]
+#[derive(serde::Serialize)]
+pub struct JsRegionDistribution {
+    chr: String,
+    start: u32,
+    end: u32,
+    n: u32,
+}
+
+#[wasm_bindgen(js_name = "RegionSet")]
+pub struct JsRegionSet {
+    region_set: RegionSet,
+}
+
+#[wasm_bindgen(js_class = "RegionSet")]
+impl JsRegionSet {
+    #[wasm_bindgen(constructor)]
+    pub fn new(regions: &JsValue) -> Result<JsRegionSet, JsValue> {
+        let regions: BedEntries = serde_wasm_bindgen::from_value(regions.clone())?;
+        let regions: Vec<Region> = regions
+            .0
+            .into_iter()
+            .map(|be| Region {
+                chr: be.0,
+                start: be.1,
+                end: be.2,
+                rest: None,
+            })
+            .collect::<Vec<Region>>();
+        let mut region_set: RegionSet = RegionSet::from(regions);
+        region_set.sort();
+        Ok(JsRegionSet { region_set })
+    }
+
+    #[wasm_bindgen(getter, js_name = "numberOfRegions")]
+    pub fn get_region(&self) -> i32 {
+        self.region_set.len() as i32
+    }
+
+    #[wasm_bindgen(getter, js_name = "meanRegionWidth")]
+    pub fn get_mean_region_width(&self) -> f64 {
+        self.region_set.mean_region_width()
+    }
+
+    #[wasm_bindgen(getter, js_name = "totalNucleotides")]
+    pub fn get_total_nucleotides(&self) -> i32 {
+        self.region_set.nucleotides_length() as i32
```

Comment:
I `nucelotides_length` right? That API seems awkward I feel like it should be the same... maybe put another way..

In javascript this is called like:
```ts
rs.totalNucleotides()
```
In the wasm bindings its called:
```rust
rs.get_total_nucleotides()
```
Then in the core implementation it seems:
```rust
rs.nucleotides_length()
```
Which are all three different APIs... maybe we can make them the same? Keep the idiomatic casing (i.e. `camelCase` in JS and `snake_case` in rust)

Reply:
@khoroshevskyi

Reply:
fixed